cypress-io/cypress
# feat(unify): allow prompts to auto-open after set time

- Closes https://cypress-io.atlassian.net/browse/UNIFY-501

This is a cleaned up version of https://github.com/cypress-io/cypress/pull/18743 with a couple of minor changes to adapt to the latest and greatest setup.


### Additional details

This recreates the functionality developed in the old desktop-gui and applies it to the runner. There's a bit of nuance to the old implementation that implies handling of more than just the 2 prompts the way they work now. I've kept that nuance.

This PR also includes some tweaks to the header that were noticed while working on this ticket. So in Launchpad:

- Header is now fixed to the top of the window
- Some z-index values have been updated for modals etc to keep things correct visually

#### Summary of code changes

This PR:

- makes project saved state available in gql
- adds a mutation to set the "prompts shown" state
- adds new project config fixtures for testing - previously in mocks, `currentProject` was populating config with a raw json object from the fixture file. The config now matches the array format that comes from `getResolvedConfigFields` so the mocks and the real app have the same format.
- moves over existing tests around prompt behavior that weren't already covered by component tests
- uses `ci1` and `orchestration1` to denote prompts for consistency
- works around some limitations of Headless UI to meet our needs for this PR, in the long run if we which libraries for this component some of the new code here can go away. This is what a lot of the `forceOpen` stuff relates to
- modifies some header styles to avoid bad text wrapping
- improves the accessible name for the profile/log out menu

### How has the user experience changed?

If it's been more than 4 days since a project was first opened, and there is not yet a cloud project ID set up, the prompt encouraging the user to set up CI will appear when the runner is launched.

### How to test

1. Remove the `projectId` from the cypress.json file of the project that you plan to test.
2. Open the App (not launchpad) for that project
3. If this projects was first opened more than 4 days ago, the prompt should appear.
4. Dismiss the prompt in any fashion.
5. Refresh the page.
6. The prompt should not appear again.

To change the saved state of the project, look for the `state.json` files nested under `/Library/Application Support/Cypress/cy/staging/projects/`

To play around with other things, modify the prompts here in `HeaderBarContent.vue`

```js
const prompts = sortBy([
  {
    slug: 'ci1',
    interval: interval('4 days'),
    noProjectId: true,
  },
  {
    slug: 'orchestration1',
    noProjectId: true,
  },
], 'interval')
```

### PR Tasks
